src/drivers/github.go

```diff
@@ -22,8 +30,32 @@ type githubCodeHostingDriver struct {
 	repository string
 }
 
-func (d *githubCodeHostingDriver) CanBeUsed(driverType string) bool {
-	return driverType == "github" || d.hostname == "github.com"
+// LoadGithub provides a GitHub driver instance if the given repo configuration is for a Github repo,
+// otherwise nil.
+func LoadGithub(config GithubConfig) CodeHostingDriver {
+	driverType := config.GetCodeHostingDriverName()
+	originURL := config.GetRemoteOriginURL()
+	hostname := helpers.GetURLHostname(originURL)
+	configuredHostName := config.GetCodeHostingOriginHostname()
+	if configuredHostName != "" {
+		hostname = configuredHostName
+	}
+	if driverType != "github" && hostname != "github.com" {
+		return nil
+	}
+	repositoryParts := strings.SplitN(helpers.GetURLRepositoryName(originURL), "/", 2)
+	if len(repositoryParts) != 2 {
+		return nil
```

Comment:
Have you found a case where it get's that far?

Reply:
We should never hit this **in theory**. Not checking this feels too much like recklessly ignoring a possible error condition. Better safe than sorry.